cmake/Utils.cmake

```diff
@@ -59,17 +59,24 @@ MACRO(ADD_TARGET_PROPERTY TARGET PROPERTY VALUE)
     ENDIF()
 ENDMACRO(ADD_TARGET_PROPERTY)
 
-MACRO(GET_APP_VERSION APP_DIR VERSION_MAJOR VERSION_MINOR VERSION_MAINTENANCE)
-    FILE(READ "${APP_DIR}/version" VERSION_RAW)
-    STRING(REGEX MATCHALL "[0-9]" VERSION_PARTS ${VERSION_RAW})
-    LIST(GET VERSION_PARTS 0 ${VERSION_MAJOR})
-    LIST(GET VERSION_PARTS 1 ${VERSION_MINOR})
-    LIST(GET VERSION_PARTS 2 ${VERSION_MAINTENANCE})
+MACRO(GET_APP_VERSION GIT_DESCRIBE VERSION_MAJOR VERSION_MINOR VERSION_MAINTENANCE)
+    STRING(REGEX MATCH "v([0-9]+)[.]([0-9]+)[.]([0-9]+)" GIT_DESCRIBE_MATCH ${GIT_DESCRIBE})
```

Comment:
I suppose that you can't use `\d` and `\d`? Because this will also match `0x3#2`, it doesn't enforce the dots, right?